storage/local.go

```diff
@@ -0,0 +1,131 @@
+package storage
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	"gopkg.in/src-d/core-retrieval.v0/model"
+	kallax "gopkg.in/src-d/go-kallax.v1"
+)
+
+type localRepo struct {
+	ID       kallax.ULID
+	Endpoint string
+	Status   model.FetchStatus
+}
+
+func (r *localRepo) toRepo() *model.Repository {
+	return &model.Repository{
+		ID:        r.ID,
+		Status:    r.Status,
+		Endpoints: []string{r.Endpoint},
+	}
+}
+
+type localRepoStore struct {
+	sync.RWMutex
+	repos map[kallax.ULID]*localRepo
```

Comment:
I don't know what is the specific use of this once it's running, but since use a `sync.Map`has performance implications, have we evaluate use it instead a regular map with a lock? What of the both options fix better in our usage case?

Reply:
This is not a performance bottleneck at all taking into account that we are making I/O operations with repos, but it could simplify the code in some parts, while uglify it in others (`interface{}` all the way).
Both options will work perfectly fine, but I'd rather stick with the non-`interface{}` one. What do you guys think?